### DISCUSSION TITLE
Is it possible to start UoW transaction in application sevice and commit it in another?

### DISCUSSION THREAD

**Discussion by BekAllaev** [abpframework/abp#21521] [General]
I have `PaymentApplicationService` where I create payment, after that I need to execute method in the `OrderApplicationService` which marks order for which I have made a payment as paid. I think that this should be done in boundaries of one transaction. Does ABP allows me to reach this kind of functionality?

I see that in my `PaymentApplicationService` I can have `UnitOfWorkManager` from the parent class - `ApplicationService` and  `UnitOfWorkManager` have `Begin` method which can start transaction. 

Let's say I call `Begin` method in procedure `Foo` of `PaymentApplicationService`. I want to commit this transaction in the `Foo1` procedure of `OrderApplicationService`. 

This is the place where I have doubts, I am not sure if the transaction that has started in `Foo` procedure won't be commited when `Foo` procedure is completed.

**Comment by maliming** [comment]
hi

> I have PaymentApplicationService where I create payment, after that I need to execute method in the OrderApplicationService which marks order for which I have made a payment as paid. I think that this should be done in boundaries of one transaction. Does ABP allows me to reach this kind of functionality?

Are these application services called in the same request or two requests?

Because you shouldn't call another `app service` in a `app service`.

---

Consider use Domain Service


https://abp.io/docs/latest/framework/architecture/best-practices/domain-services
https://abp.io/docs/latest/framework/architecture/domain-driven-design/domain-services

**Reply by BekAllaev** [reply]
Hello, @maliming, thanks for response. 
No I don't call one app service from another. I fire the event and in the event handler I call another app service. Are they run in one transaction? Here is the code of the first app service(take a look to the `CreateAsync`) - https://github.com/SergeyDavidovich/OMSBlazor/blob/dev/modules/src/StripeModule.Application/ApplicationServices/PaymentApplicationService.cs.

Here is the event handler - https://github.com/SergeyDavidovich/OMSBlazor/blob/dev/src/OMSBlazor.Application/EventHandlers/PaymentCreatedEventHandler.cs

**Reply by maliming** [reply]
> I fire the event and in the event handler I call another app service

Don't do this either. Consider calling `Domain Service` in your event handler.


Also see https://abp.io/docs/latest/framework/infrastructure/event-bus/local#transaction-exception-behavior

**Reply by BekAllaev** [reply]
@maliming got it, thank you for assistance. Now I see that event handling occurs in the same boundries of UoW as the code from which I fire the event.
